plugins/inputs/suricata/suricata.go

```diff
@@ -0,0 +1,229 @@
+package suricata
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+const (
+	// InBufSize is the input buffer size for JSON received via socket.
+	// Set to 10MB, as depending on the number of threads the output might be
+	// large.
+	InBufSize = 10 * 1024 * 1024
+)
+
+// Suricata is a Telegraf input plugin for Suricata runtime statistics.
+type Suricata struct {
+	sync.Mutex
+
+	Source    string `toml:"source"`
+	Delimiter string `toml:"delimiter"`
+
+	InputListener *net.UnixListener
+	Ctx           context.Context
+	Cancel        context.CancelFunc
+
+	wg sync.WaitGroup
+}
+
+// Description returns the plugin description.
+func (s *Suricata) Description() string {
+	return "Suricata stats plugin"
+}
+
+const sampleConfig = `
+  ## Data sink for Suricata stats log
+  # This is expected to be a filename of a
+  # unix socket to be created for listening.
+  # Will be overwritten if a socket or file
+  # with that name already exists.
+  source = "/var/run/suricata-stats.sock"
+  # Delimiter for flattening field keys, e.g. subitem "alert" of "detect"
+  # becomes "detect_alert" when delimiter is "_".
+  delimiter = "_"
+`
+
+// SampleConfig returns a sample TOML section to illustrate configuration
+// options.
+func (s *Suricata) SampleConfig() string {
+	return sampleConfig
+}
+
+// Start initiates background collection of JSON data from the socket
+// provided to Suricata.
+func (s *Suricata) Start(acc telegraf.Accumulator) error {
+	var err error
+	s.Lock()
+	defer s.Unlock()
```

Comment:
You don't need this mutex because Telegraf does not overlap calls to the plugin.  The goroutine we launch here can overlap calls to Gather/Stop, but all we use in it is the net.Listener which is safe to call from multiple goroutines: https://golang.org/pkg/net/#Listener

Reply:
Removed.